src/detector/implementations/aruco_marker_implementation.py

```diff
@@ -0,0 +1,280 @@
+from src.detector.api import \
+    GetDetectionParametersResponse, \
+    GetMarkerSnapshotsRequest, \
+    GetMarkerSnapshotsResponse, \
+    SetDetectionParametersRequest
+from src.common import \
+    EmptyResponse, \
+    ErrorResponse, \
+    get_kwarg
+from src.common.structures import \
+    CornerRefinementMethod, \
+    CORNER_REFINEMENT_METHOD_DICTIONARY_INT_TO_TEXT, \
+    CORNER_REFINEMENT_METHOD_DICTIONARY_TEXT_TO_INT, \
+    DetectionParameters, \
+    MarkerSnapshot, \
+    MarkerCornerImagePoint
+
+from src.common.structures.marker_status import MarkerStatus
+
+import datetime
+import logging
+from typing import Any, Callable
+import cv2.aruco
+import numpy
+
+from src.detector.implementations import abstract_marker_interface
+
+logger = logging.getLogger(__name__)
+
+class ArucoMarker(abstract_marker_interface.AbstractMarkerInterface):
+    _marker_status: MarkerStatus  # internal bookkeeping
```

Comment:
As per comment in `AbstractMarkerInterface` class, consider moving to that class. You can either make it non-private (remove the leading `_`) or add a function to get it or its data.